packages/wonder-blocks-labeled-field/src/components/__tests__/labeled-field.test.tsx

```diff
@@ -215,11 +230,240 @@ describe("LabeledField", () => {
                     </I18nInlineMarkup>
                 }
             />,
+            defaultOptions,
         );
 
         // Assert
         const description = screen.getByText("description");
         // LabelSmall has a font-size of 16px
         expect(description).toHaveStyle("font-size: 14px");
     });
+
+    describe("Attributes", () => {
+        it("should use the id prop for the base of element ids", () => {
+            // Arrange
+            const id = "example-id";
+            const label = "Label";
+            const description = "Description of the field";
+            const error = "Error message";
+            render(
+                <LabeledField
+                    field={<TextField value="" onChange={() => {}} />}
+                    id={id}
+                    label={label}
+                    description={description}
+                    error={error}
+                />,
+                defaultOptions,
+            );
+
+            // Act
+            const labelEl = screen.getByText(label);
+            const descriptionEl = screen.getByText(description);
+            const fieldEl = screen.getByRole("textbox");
+            const errorEl = screen.getByText(error);
```

Comment:
**nit:** this works well, but I wonder if we could use the `byRole` fn here, so we can ensure that we are using the correct element for referencing the error? (maybe this is covered in a different test, but wanted to see your thoughts on this).

```suggestion
            const errorEl = screen.getByRole("alert");
```

Reply:
Updated, though it may change based on the [error message implementation thread](https://github.com/Khan/wonder-blocks/pull/2339#discussion_r1792093660)